data/transactions/logic/eval.go

```diff
@@ -57,7 +57,7 @@ const MaxByteMathSize = 64
 const MaxLogSize = 1024
 
 // MaxLogCalls is the limit of total log calls during a program execution
-const MaxLogCalls = config.MaxLogCalls
+const MaxLogCalls = 32
```

Comment:
By not using the same variable, the MaxEncodedTealLogCalls becomes completely "invisible". A reasonable person might make it so that teal v6 allows more logs calls, but preserves compatibility for v5 and below.  They would not even know there is a place where we put alloc bounds on it.

Here's an entirely different proposal that I think is simple, correct, and the same as what we do elsewhere.  Notice how we set the bounds on deltas:
```
	// These bounds could be tighter, but since these values are just to
	// prevent DoS, setting them to be the maximum number of allowed
	// executed TEAL instructions should be fine (order of ~1000)
	checkSetMax(p.MaxAppProgramLen, &MaxStateDeltaKeys)
	checkSetMax(p.MaxAppProgramLen, &MaxEvalDeltaAccounts)
	checkSetMax(p.MaxAppProgramLen, &MaxAppProgramLen)
```

This is wildly bigger than it needs to be, but we're only trying to avoid crazy huge DoS messages.  So let's just set the bound on MaxLogCalls the same way.  Add:
`checkSetMax(p.MaxAppProgramLen, &MaxLogCalls)`

Now it's derived from consensus and doesn't need to be revisited.

Although we have *another* problem - nowadays, I suppose these bounds are not as obviously correct as they used to be.  You could loop, setting many keys, perhaps more that 1000.  We could use the MaxAppCost, rather than MaxAppLen though, since that's what would stop us. (Though it would need to be MaxGroupSize*MaxAppCost. because of pooling)

Reply:
Looks reasonable, done